### ISSUE TITLE
Not waiting for deployment deletion leads to flaky test results

### ISSUE THREAD

**Issue by nschhina** [istio/istio.io#7536]
To reproduce:
Deploy samples/sleep
Verify xyz...
Delete samples/sleep
Deploy samples/sleep // In a test right after previous test

export SOURCE_POD=$(kubectl get pod -l app=sleep -o jsonpath={.items..metadata.name})
// the command above can have 2 sleep deployments(1 terminating + 1 running) and
// SOURCE_POD would look something like "sleep-gibberish1 sleep-gibberish2"

now use SOURCE_POD in a curl command to verify xyz2...
TEST FAILS as SOURCE_POD is not valid

Possible fix:
Use a safe delete method that waits for deployments to terminate before proceeding to next test

**Comment by frankbu** [comment]
Yeah, we noticed this problem before and added a fix in samples.sh to deal with it: https://github.com/istio/istio.io/blob/5fbb14bedc06a5a8aeda25e9ae5c8c6538afa6f8/tests/util/samples.sh#L35.

But, that doesn't help if the sleep sample is started in a snip_ function, instead of using the startup_sleep_sample util function.

One of the framework improvements we have in mind, is that the test framework will check the status before and after every test to make sure things are cleaned up and restored to the initial state. Making sure the sleep pod is gone is another thing it should make sure of because all the docs that use it assume there is exactly one sleep pod in the namespace, so a leftover terminating one is a problem.
